exporters/metric/cortex/auth.go

```diff
@@ -0,0 +1,86 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cortex
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+)
+
+// ErrFailedToReadFile occurs when a password / bearer token file exists, but could
+// not be read.
+var ErrFailedToReadFile = fmt.Errorf("failed to read password / bearer token file")
+
+// addBasicAuth sets the Authorization header for basic authentication using a username
+// and a password / password file.
```

Comment:
We should probably add to this (and the `addBearerTokenAuth` documentation) that no action is taken if authorization already exists or is not configured for the Exporter.

Maybe something like:

```suggestion
// and a password / password file. The header value is not changed if it already exists and no action is taken
// if the Exporter is not configured with basic authorization credentials.
```

This isn't blocking.